code/modules/clothing/clothing.dm

```diff
@@ -477,6 +477,13 @@ BLIND     // can't see anything
 			. += mutable_appearance('icons/effects/item_damage.dmi', "damaged[blood_overlay_type]")
 		if(blood_DNA)
 			. += mutable_appearance('icons/effects/blood.dmi', "[blood_overlay_type]blood")
+		var/mob/living/carbon/human/M = loc

```

Comment:
I dont think you need to do this.
I think you can just use the layer var on the accessory, and pass that to the accessory's mutable appearance.
so long as the layer isn't negative it'll layer "in-world" as opposed to "in-parent-object" when it is negative.